packages/jaeger-ui/src/components/TracePage/CriticalPath/index.tsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2023 The Jaeger Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import memoizeOne from 'memoize-one';
+import { Span, Trace, criticalPathSection } from '../../../types/trace';
+import getChildOfSpans from './utils/getChildOfSpans';
+import findLastFinishingChildSpanId from './utils/findLastFinishingChildSpanId';
+import findRootSpanId from './utils/findRootSpanId';
+import sanitizeOverFlowingChildren from './utils/sanitizeOverFlowingChildren';
+
+export const computeCriticalPath = (
+  spanMap: Map<string, Span>,
+  spanId: string,
+  criticalPath: criticalPathSection[],
+  spawnTime?: number
+): criticalPathSection[] => {
+  // spawnTime refers to spawn synchronization event of current span
+  const currentSpan: Span = spanMap.get(spanId)!;
+
+  const lastFinishingChildSpanId = findLastFinishingChildSpanId(spanMap, currentSpan, spawnTime);
+  let spanCriticalSection: criticalPathSection;
+
+  if (lastFinishingChildSpanId) {
+    const lfcSpan = spanMap.get(lastFinishingChildSpanId)!;
```

Comment:
why do you return `lastFinishingChildSpanId` (an ID) above instead of the actual span, if you're going to need the span anyway?